django_redis/client/default.py

```diff
@@ -566,12 +566,12 @@ def make_pattern(self, pattern, version=None, prefix=None):
         return CacheKey(self._backend.key_func(pattern, prefix, version))
 
     def close(self, **kwargs):
-        if getattr(settings, "DJANGO_REDIS_CLOSE_CONNECTION", False):
-            for i in range(len(self._clients)):
-                client = self._clients[i]
-                if client is not None:
-                    client.connection_pool.disconnect()
-                self._clients[i] = None
+        close_flag = self._options.get(
+            "CLOSE_CONNECTION", getattr(settings, "DJANGO_REDIS_CLOSE_CONNECTION", True)
+        )
+        if close_flag:
+            [client.connection_pool.disconnect() for client in self._clients if client]
```

Comment:
This is bad for readability, the good old for is easier to read and equivalent

Reply:
Agree, compound statements can be really bad for readability (just like most one-liners), still IMO list comprehension  is a great tool much appreciated for  brevity and  expressiveness when coded  in its simplest form  -> [do <smt> for <each> in iterable if <condition>] <- love it 🥇 
(here I'm trying to style code in the famous "pythonic way")
I will change it following your request anyway as I am not the mantainer.

PS
what about changing client.connection_pool.disconnect() with something like self.disconnect(idx) that follows the connect connect(idx) approach of delegating  to the connection factory connect/disconnect actions based on the configured client class? will propose an easy implementation based on this approach (there is always the opportunity to hear from your feedback :)

Reply:
this is not pythonic in my honest opinion, it would be if you would have used it as an array after, otherwise it is just writing less code and making it difficult to read, I invite you to read [the zen of python](https://www.python.org/dev/peps/pep-0020/).
In this regard, the first seven points describe the reason why I think the is not pythonic.

Reply:
done